Análise de Vendas

In [14]:
# Importação das bibliotecas
import numpy as np
import pandas as pd
import datetime as dt

Fazendo o carregamento do arquivo e as verificações iniciais dos dados.

In [15]:
# Carrega o dataset
dados = pd.read_csv('vendas.csv')

In [16]:
# analisando o shape do arquivo
dados.shape

(9700, 11)

In [17]:
# analisando uma amostra dos dados
dados.head()

ID_Pedido Data_Pedido ID_Cliente   Segmento           Pais  \
0  CA-2017-152156  08/11/2017   CG-12520   Consumer  United States   
1  CA-2017-152156  08/11/2017   CG-12520   Consumer  United States   
2  CA-2017-138688  12/06/2017   DV-13045  Corporate  United States   
3  US-2016-108966  11/10/2016   SO-20335   Consumer  United States   
4  US-2016-108966  11/10/2016   SO-20335   Consumer  United States   

            Cidade      Estado       ID_Produto        Categoria SubCategoria  \
0        Henderson    Kentucky  FUR-BO-10001798        Furniture    Bookcases   
1        Henderson    Kentucky  FUR-CH-10000454        Furniture       Chairs   
2      Los Angeles  California  OFF-LA-10000240  Office Supplies       Labels   
3  Fort Lauderdale     Florida  FUR-TA-10000577        Furniture       Tables   
4  Fort Lauderdale     Florida  OFF-ST-10000760  Office Supplies      Storage   

   Valor_Venda  
0     261.9600  
1     731.9400  
2      14.6200  
3     957.5775  
4   22368.0000

In [18]:
# Colunas do conjunto de dados
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9700 entries, 0 to 9699
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   ID_Pedido     9700 non-null   object 
 1   Data_Pedido   9700 non-null   object 
 2   ID_Cliente    9700 non-null   object 
 3   Segmento      9700 non-null   object 
 4   Pais          9700 non-null   object 
 5   Cidade        9700 non-null   object 
 6   Estado        9700 non-null   object 
 7   ID_Produto    9700 non-null   object 
 8   Categoria     9700 non-null   object 
 9   SubCategoria  9700 non-null   object 
 10  Valor_Venda   9700 non-null   float64
dtypes: float64(1), object(10)
memory usage: 833.7+ KB


In [19]:
# Resumo estatístico da coluna valor de venda
dados['Valor_Venda'].describe()

count    9.700000e+03
mean     8.352892e+04
std      3.327765e+05
min      9.900000e-01
25%      3.990000e+01
50%      3.516800e+02
75%      2.403600e+04
max      1.119997e+07
Name: Valor_Venda, dtype: float64

In [20]:
# Verificando se há registros duplicados. No caso não existem
dados[dados.duplicated()]

Empty DataFrame
Columns: [ID_Pedido, Data_Pedido, ID_Cliente, Segmento, Pais, Cidade, Estado, ID_Produto, Categoria, SubCategoria, Valor_Venda]
Index: []

In [22]:
# Verificando quantos valores ausentes existem em cada coluna. No caso nenhum
dados.isnull().sum()

ID_Pedido       0
Data_Pedido     0
ID_Cliente      0
Segmento        0
Pais            0
Cidade          0
Estado          0
ID_Produto      0
Categoria       0
SubCategoria    0
Valor_Venda     0
dtype: int64

## Pergunta de Negócio 1 (nível easy):

### Qual cidade onde foi feito o maior faturamento com vendas de produtos da categoria 'Office Supplies'?

## Pergunta de Negócio 2 (nível easy):

### Qual o total de vendas segmentado por dia (data do pedido)?

## Pergunta de Negócio 3 (nível easy):

### Qual o total de vendas por estado?

## Pergunta de Negócio 4 (Nível easy):

### Quais são as 10 cidades onde a loja teve maior faturamento em vendas?

## Pergunta de Negócio 5 (nível médio):

### Qual cliente da cidade de Los Angeles geral mais faturamento (gastou mais dinheiro na loja)?

## Pergunta de Negócio 6 (Nível médio):

### Qual o total de vendas por Segmento e por ano?

## Pergunta de Negócio 7 (Nível Rambo):

Os gestores da empresa estão pensando em considerar oferecer descontos conforme o valor dos pedidos. Basicamente, pedidos inferiores a 1000 teriam desconto de 10% e acima deste valor um desconto de 15%. Desta forma, considerando os dados que temos, eles gostariam de fazer uma simulação, para ter ideia de quantos pedidos receberiam os desconto de 15%.

### Quantas vendas receberiam desconto de 10% e de 15% seguindo esta regra?

## Pergunta de Negócio 8 (Nível Verdão):

### Considerando que a empresa decida oferecer os descontos (de 10% ou 15%) citados na questão anterior. Qual seria a queda percentual no faturamento da empresa?